spec/integration/overrides_email_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "OAuth2 Overrides Email", type: :request do
+  fab!(:initial_email) { "initial@example.com" }
+  fab!(:new_email) { "new@example.com" }
+  fab!(:user) { Fabricate(:user, email: initial_email) }
+  fab!(:uac) { UserAssociatedAccount.create!(user: user, provider_name: "oauth2_basic", provider_uid: "12345") }
+
+  before do
+    SiteSetting.oauth2_enabled = true
+    SiteSetting.oauth2_callback_user_id_path = "uid"
+    SiteSetting.oauth2_fetch_user_details = false
+    SiteSetting.oauth2_email_verified = true
+
+    OmniAuth.config.test_mode = true
+    OmniAuth.config.mock_auth[:oauth2_basic] = OmniAuth::AuthHash.new(
+      provider: 'oauth2_basic',
+      uid: '12345',
+      info: OmniAuth::AuthHash::InfoHash.new(
+        email: new_email
+      ),
+      extra: {
+        raw_info: OmniAuth::AuthHash.new(
+          email_verified: true
+        )
+      },
+      credentials: OmniAuth::AuthHash.new
+    )
+  end
+
+  it 'updated user email if enabled' do
```

Comment:
```suggestion
  it 'updates user email if enabled' do
```

Reply:
Also, is there an opposing test case?

Reply:
> Also, is there an opposing test case?

Right now it's built-in to this test case (I toggle the site setting halfway through). Let me split it up to make it clearer 👀

Reply:
ahaaaa